tests/std/tests/GH_000431_lex_compare_family/test.cpp

```diff
@@ -0,0 +1,569 @@
+    }
+}
+
+#ifdef __cpp_lib_concepts
+template <class Int1, class Int2, class LexCompareThreeWayFn, class Comp = compare_three_way>
+void test_three_way_algorithms_for_integrals(
+    LexCompareThreeWayFn lexicographical_compare_three_way, Comp comp = compare_three_way{}) {
+    Int1 arr1[10] = {5, 7, 3, 4, 6, 4, 7, 1, 9, 5};
+    Int2 arr2[10] = {5, 7, 3, 4, 6, 4, 7, 1, 9, 5};
+    Int2 arr3[10] = {5, 7, 3, 4, 6, 4, 7, 1, 10, 5};
+    Int2 arr4[10] = {5, 7, 2, 4, 6, 4, 7, 1, 9, 5};
+    Int2 arr5[11] = {5, 7, 3, 4, 6, 4, 7, 1, 9, 5, 4};
+    Int2 arr6[11] = {5, 7, 3, 4, 6, 4, 7, 1, 10, 5, 4};
+    Int2 arr7[11] = {5, 7, 2, 4, 6, 4, 7, 1, 9, 5, 4};
+    Int2 arr8[9]  = {5, 7, 3, 4, 6, 4, 7, 1, 9};
+    Int2 arr9[9]  = {5, 7, 3, 4, 6, 4, 7, 1, 10};
+    Int2 arr10[9] = {5, 7, 2, 4, 6, 4, 7, 1, 9};
+
+    assert(lexicographical_compare_three_way(begin(arr1), end(arr1), begin(arr2), end(arr2), comp) == 0);
+    assert(lexicographical_compare_three_way(begin(arr1), end(arr1), begin(arr3), end(arr3), comp) < 0);
+    assert(lexicographical_compare_three_way(begin(arr1), end(arr1), begin(arr4), end(arr4), comp) > 0);
+
+    assert(lexicographical_compare_three_way(begin(arr1), end(arr1), begin(arr5), end(arr5), comp) < 0);
+    assert(lexicographical_compare_three_way(begin(arr1), end(arr1), begin(arr6), end(arr6), comp) < 0);
+    assert(lexicographical_compare_three_way(begin(arr1), end(arr1), begin(arr7), end(arr7), comp) > 0);
+
+    assert(lexicographical_compare_three_way(begin(arr1), end(arr1), begin(arr8), end(arr8), comp) > 0);
+    assert(lexicographical_compare_three_way(begin(arr1), end(arr1), begin(arr9), end(arr9), comp) < 0);
+    assert(lexicographical_compare_three_way(begin(arr1), end(arr1), begin(arr10), end(arr10), comp) > 0);
+}
+
+template <class Int, class LexCompareThreeWayFn>
+void test_three_way_algorithms_for_signed_integrals(LexCompareThreeWayFn lexicographical_compare_three_way) {
+    using UInt = make_unsigned_t<Int>;
+
+    Int arr1[10]  = {5, 7, -3, 4, 6, 4, 7, 1, 9, 5};
+    Int arr2[10]  = {5, 7, +3, 4, 6, 4, 7, 1, 9, 5};
+    UInt arr3[10] = {5, 7, static_cast<UInt>(-3), 4, 6, 4, 7, 1, 9, 5};
+    UInt arr4[10] = {5, 7, static_cast<UInt>(+3), 4, 6, 4, 7, 1, 9, 5};
+
+    // Test signed
+    assert(lexicographical_compare_three_way(begin(arr1), end(arr1), begin(arr2), end(arr2), compare_three_way{}) < 0);
+    assert(lexicographical_compare_three_way(begin(arr2), end(arr2), begin(arr1), end(arr1), compare_three_way{}) > 0);
+
+    // Test unsigned
+    assert(lexicographical_compare_three_way(begin(arr3), end(arr3), begin(arr4), end(arr4), compare_three_way{}) > 0);
+    assert(lexicographical_compare_three_way(begin(arr4), end(arr4), begin(arr3), end(arr3), compare_three_way{}) < 0);
+
+    // Test comparing signed and unsigned
+    constexpr bool promoted = sizeof(Int) < sizeof(int);
+
+#ifdef __clang__ // clang doesn't allow three way comparison between int and unsigned int

```

Comment:
Is this a Clang bug that should be reported? WG21-N4892 [expr.spaceship]/4 says "If both operands have arithmetic types, or one operand has integral type and the other operand has unscoped enumeration type, the usual arithmetic conversions (7.4) are applied to the operands."

Reply:
It seems like a clang bug but I wasn't actually sure. Could someone report it? The repro is very simple:
```
#include <compare>

auto foo(int i, unsigned int j)
{
    return i <=> j;
}
```

Reply:
Thanks for the repro! Aha, Clang is correctly rejecting and MSVC is incorrectly accepting. (GCC warns because it knows about the Standard rule, but declines to enforce it with an error.)

The rule is right after the sentence that I was previously looking at: https://eel.is/c++draft/expr.spaceship#4.1 says "If a narrowing conversion (9.4.5) is required, other than from an integral type to a floating-point type, the program is ill-formed."

`int` to `unsigned int` is indeed narrowing (when the `int` isn't a literal like `42`).

Thus I believe this part of the coverage should be removed. I can validate and push changes.